### PR TITLE
code: adding first complete local installation

### PR DESCRIPTION
**Pull request by alan-mas**
Adding a new local installation of the complete indexify, extractors, cordinator, postgres and S3 locally.

Readme file with all the steps and tests have been added as well

### CODE REVIEW

File: local-installation/README.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
+# Indexify Setup Guide
+
+This guide will walk you through setting up a local development environment for Indexify using Docker, Kubernetes, and LocalStack. Follow each step carefully to ensure a successful setup. (Don.'t forget to check the important documents at the end in case you want some extra information of the setup).
```
Comment (lucasjacks0n) [inline local-installation/README.md:3]: typo do.'t

File: local-installation/README.md [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+
+## About Indexify
+
+[Indexify](https://github.com/tensorlakeai) is a reactive compute engine for extracting unstructured data and building Indexes. If you are building RAG/LLM Applications or Agents which depend on LLMs consuming data extracted from documents or video/audio, Indexify can help you to build pipelines which keeps your indexes updated in real time as data changes.. For more information and getting started, visit the [Tensorlake indexify home page](https://docs.getindexify.ai/).
```
Comment (lucasjacks0n) [inline local-installation/README.md:7]: remove extra period

File: local-installation/README.md (hunk 1 above)
Comment (lucasjacks0n) [reply local-installation/README.md:3]: can you also rephrase it to "extra information about the setup"

File: local-installation/README.md [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+## Prerequisites
+
+- [Docker Desktop](https://www.docker.com/products/docker-desktop) installed and running
+- Kubernetes enabled in Docker Desktop (instructions [here](https://docs.docker.com/desktop/kubernetes/))
+- [AWS CLI](https://aws.amazon.com/cli/) installed
+- [LocalStack](https://docs.localstack.cloud/getting-started/) installed
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) installed
+- [Helm](https://helm.sh/docs/intro/install/) installed
+- [Python](https://www.python.org/downloads/) installed
+- [Git](https://git-scm.com/downloads) installed
```
Comment (lucasjacks0n) [inline local-installation/README.md:18]: remove git as a dependency also, people who don't have git/know how to clone a repo are probably not the people using our project

File: local-installation/README.md [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+Ensure Docker Desktop is running. You should be logged in and able to run Kubernetes and Docker commands. Follow these steps to check and confirm Docker Desktop is running:
+
+### Open Docker Desktop:
+
+- **On Windows**, click on the Docker icon in the Start menu.
+- **On macOS**, open Docker from the Applications folder.
+
+### Verify Docker is Running:
+
+- Look for the Docker whale icon in your system tray (Windows) or menu bar (macOS). It should indicate that Docker is running.
```
Comment (lucasjacks0n) [inline local-installation/README.md:43]: remove verify docker is running step

File: local-installation/README.md [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+
+- [Docker Desktop](https://www.docker.com/products/docker-desktop) installed and running
+- Kubernetes enabled in Docker Desktop (instructions [here](https://docs.docker.com/desktop/kubernetes/))
+- [AWS CLI](https://aws.amazon.com/cli/) installed
+- [LocalStack](https://docs.localstack.cloud/getting-started/) installed
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) installed
+- [Helm](https://helm.sh/docs/intro/install/) installed
+- [Python](https://www.python.org/downloads/) installed
+- [Git](https://git-scm.com/downloads) installed
+- [Visual Studio Code](https://code.visualstudio.com/) (optional but highly recommended)
```
[the remark this answers is not in the document]
Comment (alan-mas) [reply local-installation/README.md:19]: I put it as a optional (inside the parentesis). IMO keep it there is good as it is a pretty good tool, and we are saying that it is only an optional one, but highly recommended.

File: local-installation/README.md [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+
+### Enable Kubernetes:
+
+1. Open Docker Desktop and go to the Settings or Preferences menu.
+2. Navigate to the "Kubernetes" tab.
+3. Check the box to enable Kubernetes and apply the changes. Docker will take a few minutes to start the Kubernetes cluster.
+
+### Confirm Docker and Kubernetes:
+
+1. Open a terminal or command prompt.
```
Comment (lucasjacks0n) [inline local-installation/README.md:54]: can you specify the minimum versions required or which version the user should see?

File: local-installation/README.md (hunk 3 above)
Comment (alan-mas) [reply local-installation/README.md:18]: I always try to add as much information as I can haha so everyone can just go there and do it by itself (lets say a Project manager that was just hanging around) and I do not think is hurting anyone, but I can remove if you want

File: local-installation/README.md [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+pip install indexify
+```
+
+Open the file `indexify-testing.ipynb` in Visual Studio and run the code blocks one by one.
+
+By following these steps, you should have a fully functional local development environment for Indexify. If you encounter any issues, refer to the documentation or seek help from the community.
+
+## Important Documents
+
+For extra documentation and info, refer to the following links:
```
Comment (lucasjacks0n) [inline local-installation/README.md:328]: I like this section for extra documentation  👍

**Review by lucasjacks0n — CHANGES_REQUESTED** [review]
a few comments on readme

File: local-installation/README.md (hunk 4 above)
Comment (alan-mas) [reply local-installation/README.md:43]: Same as above, I can do it in case you want but I think it is good to have it so everyone can give a try

Comment (lucasjacks0n) [reply local-installation/README.md:43]: I think in this case it is better to be more concise and users should already know how to use docker

File: local-installation/README.md (hunk 3 above)
Comment (lucasjacks0n) [reply local-installation/README.md:18]: yeah I would still remove it

File: local-installation/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (lucasjacks0n) [reply local-installation/README.md:19]: yeah I would still say remove this, not needed, if a user already has an IDE they should be fine, or if they don't they probably won't be setting this up

File: local-installation/README.md [hunk 8]
```diff
@@ -0,0 +1,337 @@
...
+## About Indexify
+
+[Indexify](https://github.com/tensorlakeai) is a reactive compute engine for extracting unstructured data and building Indexes. If you are building RAG/LLM Applications or Agents which depend on LLMs consuming data extracted from documents or video/audio, Indexify can help you to build pipelines which keeps your indexes updated in real time as data changes.. For more information and getting started, visit the [Tensorlake indexify home page](https://docs.getindexify.ai/).
+
+## Prerequisites
+
+- [Docker Desktop](https://www.docker.com/products/docker-desktop) installed and running
+- Kubernetes enabled in Docker Desktop (instructions [here](https://docs.docker.com/desktop/kubernetes/))
+- [AWS CLI](https://aws.amazon.com/cli/) installed
+- [LocalStack](https://docs.localstack.cloud/getting-started/) installed
```
[the remark this answers is not in the document]
Comment (alan-mas) [reply local-installation/README.md:14]: Yes I can give a try on it :)

File: local-installation/README.md (hunk 2 above)
Comment (alan-mas) [reply local-installation/README.md]: Done :)

File: local-installation/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (alan-mas) [reply local-installation/README.md]: Done :)

File: local-installation/README.md (hunk 3 above)
Comment (alan-mas) [reply local-installation/README.md]: Done :)

File: local-installation/README.md (hunk 4 above)
Comment (alan-mas) [reply local-installation/README.md]: Done :)

File: local-installation/indexify-helm-chart/charts/ingress-nginx-4.0.6.tgz
Comment (lucasjacks0n) [inline local-installation/indexify-helm-chart/charts/ingress-nginx-4.0.6.tgz:1]: do we need this file?

File: local-installation/indexify-helm-chart/templates/NOTES.txt
Comment (lucasjacks0n) [inline local-installation/indexify-helm-chart/templates/NOTES.txt:1]: what is this file for?

File: local-installation/README.md [hunk 9]
```diff
@@ -0,0 +1,397 @@
...
+
+This is the complete installation using kubernetes, helm and localstack, please do it in case you want to know the complete process and setup for indexify. In case you just want to go and run, please follow the new [Helm installation guide](#helm-installation)
+
+### Step 1: Clone the Repository
+
+First, clone this repository to your local machine:
+
+```sh
+git clone https://github.com/tensorlakeai/indexify-aws-deployment
+cd local-installation
```
Comment (lucasjacks0n) [inline local-installation/README.md:49]: update this to cd indexify-aws-deployment/local-installation

File: local-installation/indexify-helm-chart/templates/NOTES.txt
Comment (alan-mas) [reply local-installation/indexify-helm-chart/templates/NOTES.txt:1]: When you finish running helm, this is the file/info you see on your command line. Is just like an info file after the helm installation (it is created automatically when you run helm create command to generate a chart directory with the common files and directories used in a chart).

I left it there but I can remove it :)

File: local-installation/indexify-helm-chart/charts/ingress-nginx-4.0.6.tgz
Comment (alan-mas) [reply local-installation/indexify-helm-chart/charts/ingress-nginx-4.0.6.tgz:1]: Noup, sorry, let met remove it (this is generated everytime you run your dependencies as this is a helm dependencie for our helm)

Comment (alan-mas) [reply local-installation/indexify-helm-chart/charts/ingress-nginx-4.0.6.tgz:1]: done
